File: backend/waitingNow/src/main/java/com/waiting/waitingnow/domain/MenuVO.java
package com.waiting.waitingnow.domain;

import java.util.List;
import java.util.Map;

public class MenuVO {

    private int menuNumber;
    private String menuName;
    private int menuPrice;
    private String menuFile;
    private String menuCategory;
    private int memberNumber;
    private Map<String, List<OptionMenuVO>> menuOption;

    public int getMenuNumber() {
        return menuNumber;
    }

    public void setMenuNumber(int menuNumber) {
        this.menuNumber = menuNumber;
    }

    public String getMenuName() {
        return menuName;
    }

    public void setMenuName(String menuName) {
        this.menuName = menuName;
    }

    public int getMenuPrice() {
        return menuPrice;
    }

    public void setMenuPrice(int menuPrice) {
        this.menuPrice = menuPrice;
    }

    public String getMenuFile() {
        return menuFile;
    }

    public void setMenuFile(String menuFile) {
        this.menuFile = menuFile;
    }

    public String getMenuCategory() {
        return menuCategory;
    }

    public void setMenuCategory(String menuCategory) {
        this.menuCategory = menuCategory;
    }

    public int getMemberNumber() {
        return memberNumber;
    }

    public void setMemberNumber(int memberNumber) {
        this.memberNumber = memberNumber;
    }

    public Map<String, List<OptionMenuVO>> getMenuOption() {
        return menuOption;
    }

    public void setMenuOption( Map<String, List<OptionMenuVO>> menuOption) {
        this.menuOption = menuOption;
    }
}

File: flutter/waitingnow/lib/src/Domain/MenuVO.dart
import 'OptionMenuVO.dart';

class MenuVO {
   int? menuNumber;
   String? menuName;
   int? menuPrice;
   int? menuCount;
   String? menuFile;
   String? menuCategory;
   int? memberNumber;
   Map<String, List<OptionMenuVO>>? menuOption;

   MenuVO(){}

   Map<String, dynamic> toJson() {
      return {
         'menuNumber': menuNumber,
         'menuName': menuName,
         'menuPrice': menuPrice,
         'menuCount': menuCount,
         'menuFile': menuFile,
         'menuCategory': menuCategory,
         'memberNumber': memberNumber,
         'menuOption': menuOption?.map((key, value) => MapEntry(key, value.map((e) => e.toJson()).toList())),
      };
   }

   MenuVO.fromJson(Map<String, dynamic> json) {
      menuNumber = json['menuNumber'];
      menuName = json['menuName'];
      menuPrice = json['menuPrice'];
      menuCount = json['menuCount'];
      menuFile = json['menuFile'];
      menuCategory = json['menuCategory'];
      memberNumber = json['memberNumber'];
      menuOption = (json['menuOption'] as Map<String, dynamic>?)?.map((key, value) => MapEntry(key, (value as List<dynamic>).map((e) => OptionMenuVO.fromJson(e)).toList()));
   }
}
